internal/commands/buildpack_new.go

```diff
@@ -66,11 +70,20 @@ func BuildpackNew(logger logging.Logger, creator BuildpackCreator) *cobra.Comman
 				})
 			}
 
+			targets, warn, err := target.ParseTargets(flags.Targets)
```

Comment:
I will add the following to fix the test from my [comment](https://github.com/buildpacks/pack/pull/1921/files#r1342753791)

```golang
var targets []dist.Target
if len(stacks) == 0 {
	targets, err = target.ParseTargets(flags.Targets, logger)
	if err != nil {
		return err
	}
}
```

Also, I will suggest to change the signature of `target.ParseTargets` to:

```golang
func ParseTargets(t []string, logger logging.Logger) (targets []dist.Target, err error) 
```

You can warn the messages according and avoid the new *warn.Warn* structure